tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -227,6 +235,30 @@ static std::string removeExt(const std::string &FileName) {
 
 static ExitOnError ExitOnErr;
 
+#ifdef LLVM_SPIRV_HAVE_SPIRV_TOOLS
+/// Stream buffer that captures written data into a vector and allows reading
+/// the data back as an array of uint32_t's.
+class StreambufToArray : public std::streambuf {
+public:
+  const uint32_t *data() const {
+    return reinterpret_cast<const uint32_t *>(v.data());
+  }
+
+  size_t size() const { return v.size() / sizeof(uint32_t); }
+
+protected:
+  std::streamsize xsputn(const char *s, std::streamsize count) override {
+    for (std::streamsize I = 0; I < count; I++) {
+      v.push_back(s[I]);
+    }
+    return count;
+  }
+
+private:
+  std::vector<char> v;
```

Comment:
maybe it can be named `buffer` or something else more descriptive instead?

Reply:
Also should variable names be in the upper case or does it not apply to members?

Reply:
> Should we add any test for this?

Ideally yes, but it's not trivial to do so, because the test should only run when llvm-spirv was built with the new configuration option enabled.

A related question is whether we should add another build to the CI for the new build configuration?

> maybe it can be named buffer or something else more descriptive instead?

Sure, I'll change the name.

Reply:
> 



> Ideally yes, but it's not trivial to do so, because the test should only run when llvm-spirv was built with the new configuration option enabled.

> A related question is whether we should add another build to the CI for the new build configuration?

Is there some `REQUIRES` guards in tests already available? Yeah I imagine if we want to run this in CI we might need a new CI job or something? Although it would only run one test I imagine which is a bit wasteful... but if we want clang to use this it should be tested somewhere.... So maybe we can just extend an existing CI to run this too?

Reply:
> Is there some REQUIRES guards in tests already available?

There's not, but I have added one.